brain/compartmentReportReader.h

```diff
@@ -0,0 +1,243 @@
+/* Copyright (c) 2017, EPFL/Blue Brain Project
+ *                     Juan Hernando <juan.hernando@epfl.ch>
+ *                     Mohamed-Ghaith Kaabi <mohamed.kaabi@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include "types.h"
+#include <future>
+
+namespace brain
+{
+namespace detail
+{
+class CompartmentReportMapping;
+class CompartmentReportReader;
+class CompartmentReportFrame;
+class CompartmentReportView;
+}
+
+/**
+ * Compartment report meta data.
+ * Meta data for the CompartmentReportReader.
+ */
+struct CompartmentReportMetaData
+{
+    /** The  start time of the report */
+    float startTime = 0;
+
+    /** The  end time of the report */
+    float endTime = 0;
+
+    /** The sampling time interval of the report*/
+    float timeStep = 0;
+
+    /** The  time unit of the report */
+    std::string timeUnit;
+
+    /** The  data unit of the report */
+    std::string dataUnit;
+};
+
```

Comment:
This is not needed. All the types are move only or have default constructors which are noexcept when applicable.